### ISSUE TITLE
[Core] S3 work_dir url with space causes task to hang in pending state

### ISSUE THREAD

**Issue by larroy** [ray-project/ray#40546]
### What happened + What you expected to happen

submitted a job with working-dir containing a s3 url with a space causes the task to hang in Pending state.

### Versions / Dependencies

2.7.1

### Reproduction script

```
ray job submit -v --working-dir "s3://bucket/key with space.zip" -- "ls -l"
```

### Issue Severity

None

**Comment by larroy** [comment]
Bug probably here: https://github.com/ray-project/ray/blob/master/python/ray/_private/runtime_env/context.py#L72

**Comment by anyscalesam** [comment]
@alexeykudinkin can you please triage this Ray Jobs issue?

**Comment by akshay-anyscale** [comment]
Reassigning to core team since it's a runtime env issue

**Comment by anyscalesam** [comment]
@larroy could you submit a PR with fix; should be straightforward and we can review.

**Comment by y-abe** [comment]
I've identified a potential solution for the issue at hand. By adding a space to `disallowed_chars`, I was able to resolve the problem on my machine.

https://github.com/ray-project/ray/blob/0f2a32778307ee02a193357a94c24933e5e191d4/python/ray/_private/runtime_env/packaging.py#L210

If this aligns with your expectations, I'd be happy to make a pull request. Please let me know if this is acceptable to you.

**Comment by rkooo567** [comment]
Maybe @jjyao already fixed the issue. He's going to verify it.

**Comment by jjyao** [comment]
Fixed by #46767

**Comment by larroy** [comment]
yeah "FIXED"
